File: xml-performance/test-drive-xml-main-runner/src/test/java/org/jesperancinha/test/drive/books/XmlTestRunnerTest.java
package org.jesperancinha.test.drive.books;

import java.util.Random;

import org.junit.Test;

public class XmlTestRunnerTest {
	private static final Random random = new Random();

	@Test
	public void testMain() throws Exception {
		final String testFile = "testFile" + Math.random();
		final String testIteration = "" + random.nextInt();
		final String[] args = new String[] { //
				"-n", //
				testIteration, //
				"-b", //
				testFile, //
				"--noop" //
		};
		XmlTestRunner.main(args);
	}

	@Test
	public void testMain_aliases() throws Exception {
		final String testFile = "testFile" + Math.random();
		final String testIteration = "" + random.nextInt();
		final String[] args = new String[] { //
				"--number-of-iterations", //
				testIteration, //
				"--book-test-file", //
				testFile, //
				"--noop" //
		};
		XmlTestRunner.main(args);
	}

}

File: jeorg-kotlin-apps/performance-parsers/xml-performance/test-drive-xml-main-runner/src/test/kotlin/org/jesperancinha/test/drive/books/XmlTestRunnerTest.kt
package org.jesperancinha.test.drive.books

import org.junit.jupiter.api.Test
import java.util.*

class XmlTestRunnerTest {
    @Test
    @Throws(Exception::class)
    fun testMain() {
        val testFile = "testFile" + Math.random()
        val testIteration = "" + random.nextInt()
        val args = arrayOf(
            "-n", 
            testIteration, 
            "-b", 
            testFile, 
            "--noop"
        )
        main(args)
    }

    @Test
    @Throws(Exception::class)
    fun testMain_aliases() {
        val testFile = "testFile" + Math.random()
        val testIteration = "" + random.nextInt()
        val args = arrayOf(
            "--number-of-iterations", 
            testIteration, 
            "--book-test-file", 
            testFile, 
            "--noop"
        )
        main(args)
    }

    companion object {
        private val random = Random()
    }
}